app/javascript/hooks/useTranslate.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * This hook is for using an older version of Google Translate
+ * that requires work to ensure the timing of script loading and event
+ * dispatching is correct. Essentially, we inject a Google Translate script that adds
+ * a <select> element to the dom. That <select> element is a dropdown of different languages
+ * that we can machine translate the page with. We hide that <select> with css.
+ * Once that <select> element is rendered (even though hidden), we use the language
+ * set in the url to dispatch an event setting that <select>'s <option>
+ * to the corresponding language. This gives an experience to the end user that
+ * the page automatically got translated.
+ */
+
+import { useEffect } from "react"
+import useLanguageStringInUrl from "./useLanguageStringInUrl"
+import usePollElementRender from "./usePollElementRender"
+import useScript from "./useScript"
+const tWindow = window as any
+const languageMap = {
+  zh: "zh-TW",
+}
+
+const initGoogleTranslate = () => {
+  // eslint-disable-next-line no-new
+  new tWindow.google.translate.TranslateElement(
+    {
+      pageLanguage: "en",
+      includedLanguages: "en,es,tl,zh-TW",
+      layout: tWindow.google.translate.TranslateElement.InlineLayout.HORIZONTAL,
+      autoDisplay: false,
+      multilanguagePage: true,
+    },
+    "google_translate_element"
+  )
+}
+
+tWindow.initGoogleTranslate = initGoogleTranslate
+
+const useTranslate = () => {
+  const languageInRoute = useLanguageStringInUrl()
```

Comment:
wondering if this hook is necessary. what if we just call `getCurrentLanguage()` here. afaik changing languages currently results in a page load.

Reply:
I guess i don't understand why checking for the language code needs to be a separate hook. How will that functionality be used again? What happens if we move that state here? Also wondering if we need to run any translation code if the page is english.

If a user clicks on a language link, the browser will navigate to a new page so seems that would reload everything (housing portal is not a SPA). For other non-page loading scenarios, I'm wondering if we may need additional functionality. For example, at some point we will support overriding of preference descriptions which are loaded from a separate request from the rest of the listing data. When those get loaded will the machine translations magically work or we'll somehow have to call google translate again? Sounds like a fun run race condition.